docker-compose-rule-junit4/src/main/java/com/palantir/docker/compose/DockerComposeRule.java

```diff
@@ -148,19 +148,23 @@ public void before() throws IOException, InterruptedException {
         }
         upDockerCompose.up();
 
-        logCollector().startCollecting(dockerCompose());
         log.debug("Waiting for services");
         new ClusterWait(ClusterHealthCheck.nativeHealthChecks(), nativeServiceHealthCheckTimeout())
                 .waitUntilReady(containers());
         clusterWaits().forEach(clusterWait -> clusterWait.waitUntilReady(containers()));
         log.debug("docker-compose cluster started");
+
+        logCollector().startCollecting(dockerCompose());
```

Comment:
What if a docker container fails to start up? Won't the log collector not collect _anything_ for that container? This would be super hard to debug :(

Reply:
Yeah this a downside, but I'm not sure how it can be mitigated. The issue is that you can't follow logs across container restarts.

An alternative is that we don't do any log collection until the end (this was my original implementation in #244). But that can cause the log collection to take a long time if your containers have generated a lot of logs. And we lose the benefit of having logs while the container is still running.